scripts/ldap/base_configure_ldap.py

```diff
@@ -0,0 +1,155 @@
+
+os.system('chgrp ldap /etc/openldap/certs/password')
+
+if LDAPADMINPASS == None :
+    if defaultpasswd == True :
+        LDAPADMINPASS = defaultpasswds
+    else :
+        LDAPADMINPASS = input("Enter password for LDAP IAM admin ")
+
+if ROOTDNPASSWORD == None :
+    if defaultpasswd == True :
+        ROOTDNPASSWORD = defaultpasswds
+    else :
+        ROOTDNPASSWORD = input("Enter password for LDAP RootDN ")
+
+cmd = 'slappasswd -s ' + ROOTDNPASSWORD
+pwd = os.popen(cmd).read()
+pwd.replace('/','\/')
+#restart slapd post cleanup
+os.system('systemctl restart slapd')
+
+# Open a connection
+l = ldap.initialize("ldapi:///")
+
+# Bind/authenticate with a user with apropriate rights to add objects
+l.simple_bind_s("cn=admin,dc=seagate,dc=com",ROOTDNPASSWORD)
+
+
+l.sasl_non_interactive_bind_s('EXTERNAL')
+dn="olcDatabase={0}config,cn=config"
+mod_attrs = [( ldap.MOD_REPLACE, 'olcRootDN', [b'cn=admin,cn=config'] )]
+l.modify_s(dn,mod_attrs)
+
+mod_attrs = [( ldap.MOD_REPLACE, 'olcRootPW', pwd.encode("utf-8") )]
+l.modify_s(dn,mod_attrs)
+
+mod_attrs = [( ldap.MOD_REPLACE, 'olcAccess', [b'{0}to * by dn.base="gidNumber=0+uidNumber=0,cn=peercred,cn=external,cn=auth" write by self write by * read'] )]
+l.modify_s(dn,mod_attrs)
+
+dn="olcDatabase={2}mdb,cn=config"
+mod_attrs = [( ldap.MOD_REPLACE, 'olcSuffix', [b'dc=seagate,dc=com'] )]
+l.modify_s(dn,mod_attrs)
+
+mod_attrs = [( ldap.MOD_REPLACE, 'olcRootDN', [b'cn=admin,dc=seagate,dc=com'] )]
+l.modify_s(dn,mod_attrs)
+
+mod_attrs = [( ldap.MOD_ADD, 'olcDbMaxSize', [b'10737418240'] )]
+l.modify_s(dn,mod_attrs)
+
+mod_attrs = [( ldap.MOD_REPLACE, 'olcRootPW', pwd.encode("utf-8") )]
+l.modify_s(dn,mod_attrs)
+
+mod_attrs = [( ldap.MOD_REPLACE, 'olcAccess', [b'{0}to attrs=userPassword by self write by dn.base="cn=admin,dc=seagate,dc=com" write by anonymous auth by * none'] )]
+l.modify_s(dn,mod_attrs)
+
+mod_attrs = [( ldap.MOD_ADD, 'olcAccess', [b'{1}to * by dn.base="cn=admin,dc=seagate,dc=com" write by self write by * none'] )]
+l.modify_s(dn,mod_attrs)
+
+mod_attrs = [( ldap.MOD_REPLACE, 'olcAccess', [b'{0}to attrs=userPassword by self write by dn.base="cn=sgiamadmin,dc=seagate,dc=com" write by anonymous auth by * none'] )]
+l.modify_s(dn,mod_attrs)
+
+mod_attrs = [( ldap.MOD_ADD, 'olcAccess', [b'{1}to * by dn.base="cn=sgiamadmin,dc=seagate,dc=com" write by self write by * none'] )]
+l.modify_s(dn,mod_attrs)
+
+l.unbind_s()
+
+os.system('systemctl restart slapd')
+
+#add_s - init.ldif
+# Open a connection
+l = ldap.initialize("ldapi:///")
+
+# Bind/authenticate with a user with apropriate rights to add objects
+l.simple_bind_s("cn=admin,dc=seagate,dc=com",ROOTDNPASSWORD)
+
+dn="dc=seagate,dc=com"
+add_record = [
+ ('dc', [b'seagate'] ),
+ ('o', [b'seagate'] ),
+ ('description', [b'Root entry for seagate.com.']),
+ ('objectClass', [b'top',b'dcObject',b'organization'])
+]
+l.add_s(dn, add_record)
```

Comment:
can you catch exception of these ldap api calls and print message with detail about step that failed

Reply:
Fixed